src/test/messageHandlers/JSONHandlerTest.java

```
@@ -0,0 +1,39 @@
+package messageHandlers;
+
+import clientHandlers.JSONExtractor;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class JSONHandlerTest{
+    JSONExtractor jsonEx;
+    String testMessageOpen;
+    String testMessageClose;
+
+    @BeforeEach
+    void setUp() {
+        jsonEx = new JSONExtractor();
+        testMessageOpen = "{\"action\": {\"id\": 1, \"type\": \"openLock\"}}";
+        // testMessage is the example message on the message board in Basecamp.
+        testMessageClose = "{\"action\": {\"id\": 2, \"type\": \"closeLock\"}}";
+    }
+
+    @Test
+    void testMessageHandlerLock() {
+        assertEquals("openLock",jsonEx.handleMessage(testMessageOpen).getAction());
+        assertEquals("closeLock",jsonEx.handleMessage(testMessageClose).getAction());
+    }
+
+    @Test
+    void testPeripheralID() {
+        assertEquals(1,jsonEx.handleMessage(testMessageOpen).getTargetId());
+        assertEquals(2,jsonEx.handleMessage(testMessageClose).getTargetId());
+    }
+
+    @Test
+    void testNOP() {
+        assertEquals("NOP",jsonEx.handleMessage("Some garbage String").getAction());
```

I think it is better to throw an exception in this case.